File: scala/tour/src/main/scala/oo/Enums.scala
package oo

/**
  * Created by andersc on 5/14/17.
  */
object WeekDay extends Enumeration {
  type WeekDay = Value

  val Mon, Tue, Wed, Thu, Fri, Sat, Sun = Value
}

object TestEnums {
  import WeekDay._

  def isWorkingDay(d: WeekDay) = {
    ! (d == Sat || d == Sun)
  }

  def main(args: Array[String]): Unit = {
    WeekDay.values foreach (d => println(d.toString))
  }
}

File: c/tour/effective/enums.c
#include <stdio.h>
#include <stdlib.h>

enum day { sun, mon, tue, wed, thu, fri, sat };
enum cardinal_points { north = 0, east = 90, south = 180, west = 270 };

enum seasons: unsigned short {
    spring = 1,
    summer,
    autumn,
    winter
};

int main() {
    enum day today = 9;
    printf("today is %d\n", today);
    return EXIT_SUCCESS;
}